analyses/selection-strategy-comparison/01-selection-strategies.rmd

```diff
@@ -0,0 +1,320 @@
+
+```{r raw clustering}
+# remove low counts
+dm_set <- exp_rsem_t[, colSums(exp_rsem_t) > 100]
+
+neighbors = 15
+rsem_umap_raw <- umap::umap(exp_rsem_t, n_neighbors = neighbors)
+
+```
+
+```{r umap plotting function}
+
+umap_plot <- function(umap_model, 
+                      metadata, 
+                      sample_id = "Kids_First_Biospecimen_ID", 
+                      comparable = TRUE,
+                      filename = NA){
+  layout <- data.frame(umap_model$layout) %>% 
+    dplyr::rename_all(.funs = gsub, pattern = "^X", replacement = "UMAP") %>%
+    tibble::rownames_to_column(var = sample_id)
+  
+  plot_data <- layout %>%
+    dplyr::left_join(metadata_df, by = sample_id) %>%
+    dplyr::filter(comparable)
+  
+  u_plot <- ggplot(plot_data, 
+                      aes(x = UMAP1, 
+                          y = UMAP2, 
+                          color = disease_type_new, 
+                          shape = RNA_library)) +
+    geom_point(alpha = 0.3) +
+    labs(color = "Disease Type",
+         shape = "Selection Method")
+  
+  if(is.character(filename)){
+    ggsave(filename, 
+           u_plot,
+           width = 11,
+           height = 7)
+  }
+  
+  return(u_plot)
+}
+
+```
+
+
+```{r plot raw}
+plot_raw <- umap_plot(rsem_umap_raw, 
+                      metadata_df,
+                      filename = file.path(output_dir, "umap_raw.png"))
+
+plot_raw
+```
+
+
+So even with more dimensions, we are not surprised to see that the polyA samples are still clustering together.
+
+## Removing Pol Ya depleted genes
+
+### Any expression at all
+As a first pass, lets see what happens when we remove genes are simply not expressed in the polyA samples (or vice versa?).
+
+
+```{r get polyA exp}
+polyA_samples <- rownames(exp_rsem_t) %in% polyA_meta_df$Kids_First_Biospecimen_ID
+polyA_exp <- exp_rsem_t[polyA_samples, ]
+
+# get genes expressed at all in all samples
+polyA_gene_filter <- colSums(polyA_exp <= 0) == 0
+
+polyA_expressed <- exp_rsem_t[, polyA_gene_filter]
+ncol(polyA_expressed)
+```
+
+```{r}
+rsem_umap_polyA <- umap::umap(polyA_expressed, n_neighbors = neighbors)
+```
+
+```{r plot polyA}
+plot_polyA <- umap_plot(rsem_umap_polyA, 
+                        metadata_df, 
+                        filename = file.path(output_dir, "umap_polyA.png"))
+plot_polyA
+```
+
+### Remove low-expressed in polyA
+
+Arbitrary cutoff of expression level 1 to start. If a genes has fpkm expression of 1 or larger in at least 10 polyA samples, we will keep it.
+```{r}
+polyA_gene_filter_high <- colSums(polyA_exp >= 1) > 10
+
+polyA_expressed_high <- exp_rsem_t[, polyA_gene_filter_high]
+ncol(polyA_expressed_high)
+```
+This seems to yield a larger gene set than removing genes with no expression in at least one polyA samples, and a very similar distribution.
+
+```{r}
+rsem_umap_polyA_high <- umap::umap(polyA_expressed_high, n_neighbors = neighbors)
```

Comment:
I put together #122 and uploaded the resulting data to see if it fixed the issue, but looks like the value of `neighbors` is still too high here.